api/tests/routes/backoffice_v3/custom_reimbursement_rule_test.py

```diff
@@ -151,3 +154,169 @@ def test_list_rules_more_than_max(self, authenticated_client):
+        offerer = offerers_factories.OffererFactory()
+        response = self.post_to_endpoint(
+            authenticated_client,
+            form={
+                "offerer": offerer.id,
+                "subcategories": ["ABO_CONCERT", "ABO_JEU_VIDEO"],
+                "rate": "12,34",
+                "start_date": self.tomorrow,
+                "end_date": self.next_year,
+            },
+        )
+        assert response.status_code == 303
+        rule = finance_models.CustomReimbursementRule.query.one()
+        assert rule.offererId == offerer.id
+        assert rule.rate == Decimal("0.1234")
+        assert rule.timespan.lower == datetime.datetime.combine(self.tomorrow, datetime.datetime.min.time())
+        assert rule.timespan.upper == datetime.datetime.combine(self.next_year, datetime.datetime.max.time())
+        assert set(rule.subcategories) == {"ABO_CONCERT", "ABO_JEU_VIDEO"}
+        assert "Tarif dérogatoire créé" in authenticated_client.get(response.location).data.decode("utf-8")
+
+    def test_create_with_invalid_offerer(self, authenticated_client):
+        response = self.post_to_endpoint(
+            authenticated_client,
+            form={
+                "offerer": 0,
+                "rate": "12,34",
+                "start_date": self.tomorrow,
+            },
+        )
+        assert response.status_code == 303
+        assert finance_models.CustomReimbursementRule.query.count() == 0
+        assert "Ce tarif dérogatoire existe déjà" in authenticated_client.get(response.location).data.decode("utf-8")
+
+    def test_create_with_no_end_date(self, authenticated_client):
+        offerer = offerers_factories.OffererFactory()
+        response = self.post_to_endpoint(
+            authenticated_client,
+            form={
+                "offerer": offerer.id,
+                "rate": "12,34",
+                "start_date": self.tomorrow,
+                "end_date": "",
+            },
+        )
+        assert response.status_code == 303
+        rule = finance_models.CustomReimbursementRule.query.one()
+        assert rule.timespan.lower == datetime.datetime.combine(self.tomorrow, datetime.datetime.min.time())
+        assert rule.timespan.upper == None
+        assert "Tarif dérogatoire créé" in authenticated_client.get(response.location).data.decode("utf-8")
+
+    def test_create_with_start_date_too_early(self, authenticated_client):
+        offerer = offerers_factories.OffererFactory()
+        response = self.post_to_endpoint(
+            authenticated_client,
+            form={
+                "offerer": offerer.id,
+                "rate": "12,34",
+                "start_date": "1970-01-01",
+            },
+        )
+        assert response.status_code == 303
+        assert finance_models.CustomReimbursementRule.query.count() == 0
+        assert "Ne peut pas commencer avant demain" in authenticated_client.get(response.location).data.decode("utf-8")
```

Comment:
En fait on a des _helpers_ pour aller chercher explicitement le message flash :
```suggestion
        assert "Ne peut pas commencer avant demain" in html_parser.extract_alert(authenticated_client.get(response.location).data)
```
(pareil pour les autres)

Reply:
ok je ne les avais pas vu